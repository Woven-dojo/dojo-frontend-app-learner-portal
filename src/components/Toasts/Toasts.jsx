import React, { useContext } from 'react';
import { Toast } from '@edx/paragon';

import { ToastsContext } from './utils';

const Toasts = () => {
  const { toasts, removeToast } = useContext(ToastsContext);
  return toasts.map(({ id, content, options }) => (
    <Toast key={id} {...options} onClose={() => removeToast(id)} show>
      {content}
    </Toast>
  ));
};

export default Toasts;
