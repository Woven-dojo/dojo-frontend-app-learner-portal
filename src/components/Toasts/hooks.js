import { useContext } from 'react';
import { ToastsContext } from './utils';

export const useToast = () => {
  const context = useContext(ToastsContext);

  return {
    addToast: (content, options = {}) => {
      context.addToast(content, options);
    },
  };
};
