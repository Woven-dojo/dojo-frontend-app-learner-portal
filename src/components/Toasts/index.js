export { default as Toasts } from './Toasts';
export { default as ToastsProvider } from './ToastsProvider';

export { ToastsContext } from './utils';
