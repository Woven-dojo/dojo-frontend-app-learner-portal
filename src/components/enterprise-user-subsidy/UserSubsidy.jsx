import React, {
  createContext, useContext, useMemo, useState,
} from 'react';
import PropTypes from 'prop-types';
import { AppContext } from '@edx/frontend-platform/react';
import { Container } from '@edx/paragon';

import { LoadingSpinner } from '../loading-spinner';

import {
  useCatalogData, useLearningPathData,
} from './data/hooks';
import { LOADING_SCREEN_READER_TEXT, filterInitial, filterOptions } from './data/constants';

export const UserSubsidyContext = createContext();

const UserSubsidy = ({ children }) => {
  const { enterpriseConfig } = useContext(AppContext);
  const [catalogFilter, setCatalogFilter] = useState(filterInitial);
  const [catalogData, isLoadingCatalogData] = useCatalogData({
    enterpriseId: enterpriseConfig.uuid,
    filter: catalogFilter,
  });
  const [learningPathData, isLoadingLearningPathdata] = useLearningPathData();

  const isLoading = isLoadingCatalogData || isLoadingLearningPathdata;
  const contextValue = useMemo(
    () => {
      if (isLoading) {
        return {};
      }
      return {
        catalogData, // deprecated, please use catalog.data
        learningPathData,
        catalog: {
          data: catalogData,
          filter: {
            current: catalogFilter,
            options: filterOptions,
            set: setCatalogFilter,
          },
        },
      };
    },
    [
      isLoading,
      enterpriseConfig.uuid,
      catalogData,
      learningPathData,
      catalogFilter,
    ],
  );

  if (isLoading) {
    return (
      <Container className="py-5">
        <LoadingSpinner screenReaderText={LOADING_SCREEN_READER_TEXT} />
      </Container>
    );
  }
  return (
    <>
      {/* Render the children so the rest of the page shows */}
      <UserSubsidyContext.Provider value={contextValue}>
        {children}
      </UserSubsidyContext.Provider>
    </>
  );
};

UserSubsidy.propTypes = {
  children: PropTypes.node.isRequired,
};

export default UserSubsidy;
