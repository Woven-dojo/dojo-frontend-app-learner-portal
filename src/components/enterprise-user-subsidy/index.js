export { default as UserSubsidy, UserSubsidyContext } from './UserSubsidy';
