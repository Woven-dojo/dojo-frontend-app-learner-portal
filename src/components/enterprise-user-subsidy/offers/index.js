export { default as reducer } from './data/reducer';
