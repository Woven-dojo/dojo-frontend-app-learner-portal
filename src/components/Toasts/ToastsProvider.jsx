import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';

import { ToastsContext } from './utils';
import Toasts from './Toasts';

const ToastsProvider = ({ children }) => {
  const [toasts, setToasts] = useState([]);

  const addToast = (content, options = {}) => {
    setToasts((prevToasts) => [
      ...prevToasts,
      {
        id: uuidv4(),
        content,
        options,
      },
    ]);
  };

  const removeToast = (id) => {
    setToasts((prevToasts) => prevToasts.filter((toast) => toast.id !== id));
  };

  return (
    <ToastsContext.Provider value={{ toasts, addToast, removeToast }}>
      {children}
      <Toasts />
    </ToastsContext.Provider>
  );
};

ToastsProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ToastsProvider;
