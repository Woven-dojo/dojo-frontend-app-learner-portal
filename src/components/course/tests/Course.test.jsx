import renderer from 'react-test-renderer';
import React from 'react';
import { AppContext } from '@edx/frontend-platform/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { ToastsContext } from '../../Toasts';
import Course from '../Course';
import { UserSubsidyContext } from '../../enterprise-user-subsidy';

const ORG_SLUG = 'aperture';
const COURSE_KEY = 'think-portal';

const APP_CONTEXT = {
  enterpriseConfig: {
    name: 'Aperture Laboratories',
    slug: 'aperture',
  },
  config: {
    MARKETING_SITE_BASE_URL: 'http://example.com',
  },
};

const ACTIVE_COURSE_RUN_UUID = 'active_course_run_uuid';

jest.mock('../data/hooks', () => ({
  useAllCourseData: jest.fn(() => {
    const courseData = {
      userEnrollments: [],
      userEntitlements: [],
      userSubsidyApplicableToCourse: {
        discountType: 'absolute',
        discountValue: 100,
        expirationDate: '2022-06-01',
        startDate: '2019-06-01',
        subsidyId: 'subsidy_id',
      },
      catalog: {
        catalogList: [],
      },
      courseDetails: {
        title: 'Thinking with portals',
        advertisedCourseRunUuid: ACTIVE_COURSE_RUN_UUID,
        courseRuns: [
          {
            isEnrollable: true,
            key: 'test-course-run-key',
            pacingType: 'self_paced',
            start: '2020-09-09T04:00:00Z',
            availability: 'Current',
            uuid: ACTIVE_COURSE_RUN_UUID,
          },
          { uuid: 'course_run_uuid_1', isEnrollable: true },
          { uuid: 'non_enrollable_run_uuid_1', isEnrollable: false },
          {
            uuid: 'archived_run_uuid_1',
            availability: 'Archived',
            isEnrollable: true,
          },
        ],
        fullDescription: '<p>Description</p>',
        prerequisitesRaw: '<p>Prerequisite</p>>',
        sponsors: [
          {
            name: 'Shower Curtains',
            marketingUrl: 'sponsor-url',
          },
        ],
        skills: [],
      },
    };

    const fetchError = null;

    return [courseData, fetchError];
  }),

  useExtractAndRemoveSearchParamsFromURL: jest.fn(() => ({})),
  useCoursePartners: jest.fn(() => {
    const partners = [];
    const label = '';
    return [partners, label];
  }),

  useCourseSubjects: jest.fn(() => ({
    subjects: [],
    primarySubject: null,
  })),

  useCourseRunWeeksToComplete: jest.fn(() => {
    const weeksToComplete = 7;
    const label = 'weeks';

    return [weeksToComplete, label];
  }),

  useCourseTranscriptLanguages: jest.fn(() => {
    const languages = [];
    const label = undefined;

    return [languages, label];
  }),

  useCoursePacingType: jest.fn(() => {
    const pacingType = 'self_paced';
    const pacintTypeContent = 'Self-paced on your time';

    return [pacingType, pacintTypeContent];
  }),

  useCoursePriceForUserSubsidy: jest.fn(() => {
    const coursePrice = {
      list: 200,
      discounted: 150,
    };
    const currency = 'USD';

    return [coursePrice, currency];
  }),

  useCourseEnrollmentUrl: jest.fn(() => (
    'http://example.com'
  )),

  useTrackSearchConversionClickHandler: jest.fn(() => jest.fn()),
}));

const TOAST_CONTEXT = {
  addToast: jest.fn(),
};

const SUBSIDY_CONTEXT = {
  subscriptionLicense: {
    uuid: 'subscription_uuid',
  },
  offers: {
    offers: [],
    offersCount: 0,
  },
};

describe('Course', () => {
  afterAll(() => {
    jest.resetAllMocks();
  });

  it('should render the expected HTML', async () => {
    const url = `/${ORG_SLUG}/course/${COURSE_KEY}?course_run_key=run-key`;
    let tree;

    await renderer.act(async () => {
      tree = await renderer.create(
        <AppContext.Provider value={APP_CONTEXT}>
          <ToastsContext.Provider value={TOAST_CONTEXT}>
            <UserSubsidyContext.Provider value={SUBSIDY_CONTEXT}>
              <MemoryRouter
                initialIndex={0}
                initialEntries={[url]}
              >
                <div>
                  <Route path="/:enterpriseSlug/course/:courseKey">
                    <Course />
                  </Route>
                </div>
              </MemoryRouter>
            </UserSubsidyContext.Provider>
          </ToastsContext.Provider>
        </AppContext.Provider>,
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
