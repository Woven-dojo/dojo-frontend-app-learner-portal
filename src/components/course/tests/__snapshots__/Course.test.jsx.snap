// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course should render the expected HTML 1`] = `
<div>
  <div
    className="course-header"
  >
    <div
      className="container-mw-lg container-fluid"
    >
      <div
        className="py-4 row"
      >
        <div
          className="col-lg-7 col-12"
        >
          <div
            className="mb-4"
          >
            <h2>
              Thinking with portals
            </h2>
          </div>
          <p
            className="font-weight-bold mt-3 mb-0"
          >
            This course is not part of your company's curated course catalog.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="py-5 container-mw-lg container-fluid"
  >
    <div
      className="row"
    >
      <article
        className="col-xs-12 col-lg-12"
      >
        <div
          className="mb-5"
        >
          <ul
            className="pl-0 mb-5 course-details-sidebar"
          >
            <li
              className="d-flex align-items-start border-bottom py-3 row"
            >
              <div
                className="d-flex align-items-center col"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-tag fa-w-16 mr-3"
                  data-icon="tag"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span>
                  Price
                  :
                </span>
              </div>
              <div
                className="col"
              >
                <div
                  className="mb-2"
                >
                  <del>
                    <span
                      className="sr-only"
                    >
                      Priced reduced from:
                    </span>
                    $
                    200.00
                     
                    USD
                  </del>
                   
                  <span
                    className="sr-only"
                  >
                    Discounted price:
                  </span>
                  $
                  150.00 USD
                </div>
                <span>
                  Sponsored by 
                  Aperture Laboratories
                </span>
              </div>
            </li>
            <li
              className="d-flex align-items-start border-bottom py-3 row"
            >
              <div
                className="d-flex align-items-center col"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-user fa-w-14 mr-3"
                  data-icon="user"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span>
                  Course Type
                  :
                </span>
              </div>
              <div
                className="col"
              >
                Self-paced on your time
              </div>
            </li>
          </ul>
          <div
            className="prerequisites mb-5"
          >
            <h3>
              Prerequisites
            </h3>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Prerequisite</p>>",
                }
              }
            />
          </div>
        </div>
        <div
          className="preview-expand-component mb-5"
        >
          <h3>
            About this course
          </h3>
          <div
            className="preview-expand-body mb-3"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Description</p>",
                }
              }
            />
          </div>
          <button
            className="preview-expand-cta px-1 btn btn-link"
            disabled={false}
            id="about-this-course"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus-circle fa-w-16 mr-2 collapsible-icon"
              data-icon="plus-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            More about this course
          </button>
        </div>
        <div
          className="mb-5"
        >
          <h3>
            Sponsored by
          </h3>
          <div
            className="row no-gutters mt-3"
          >
            <div
              className="col-lg-6 mb-3"
            >
              <div
                className="mb-2"
              >
                <a
                  aria-hidden="true"
                  href="http://example.com/sponsor-url"
                  rel="noopener noreferrer"
                  tabIndex="-1"
                  target="_blank"
                >
                  <img
                    alt="Shower Curtains logo"
                  />
                </a>
              </div>
              <a
                className="pgn__hyperlink default-link standalone-link"
                href="http://example.com/sponsor-url"
                onClick={[Function]}
                rel="noopener noreferrer"
                target="_blank"
              >
                Shower Curtains
                <span
                  aria-label="in a new tab"
                  className="d-inline-block align-middle ml-2"
                  title="Opens in a new tab"
                >
                  <span
                    className="pgn__icon"
                    style={
                      Object {
                        "height": "1em",
                        "width": "1em",
                      }
                    }
                  >
                    <svg
                      aria-hidden={true}
                      fill="none"
                      focusable={false}
                      height={24}
                      role="img"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M19 19H5V5h7V3H3v18h18v-9h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
          </div>
          <p>
            The production of this course would not have been possible without the generous contributions of 
            Shower Curtains
            .
          </p>
        </div>
      </article>
    </div>
  </div>
</div>
`;
